src/test/java/org/jvnet/hudson/test/FlagRuleNoReplacementTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2020 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jvnet.hudson.test;
+
+import org.junit.AfterClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.rules.TestRule;
+
+public class FlagRuleNoReplacementTest {
+
+    private static boolean FLAG;
+
+    @BeforeClass
+    public static void start() {
+        assertFalse(FLAG);
+    }
+
+    @AfterClass
+    public static void end() {
+        assertFalse(FLAG);
+    }
+
+    @Rule
+    public TestRule flagRule = new FlagRule<Boolean>(() -> FLAG, x -> FLAG = x);
```

Comment:
As per [this](https://stackoverflow.com/a/27472025/12916) there is no slick syntax for representing field accessors. There _could_ be a factory like

```java
@Rule
public TestRule flagRule = FlagRule.field(ThisTest.class, "FLAG");
```

which would look up the field reflectively, but then this is opaque to IDEs looking to find usages etc., which does not seem to justify the slightly shorter syntax.